.github/workflows/cancel.yml

```diff
@@ -0,0 +1,49 @@
+name: Cancel redundant jobs
+
+on:
+  push:
+    branches:
+      - '!master' # always build master
+  pull_request:
+
+# workflow ids can be found via
+# https://api.github.com/repos/timberio/vector/actions/workflows
+#
+jobs:
+  cancel-tests:
+    name: Cancel redundant tests.yml workflows
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - uses: styfle/cancel-workflow-action@0.4.1
+        with:
+          workflow_id: 1463240
```

Comment:
This is capable of taking a comma separated list, but I liked the self documenting nature of separate steps.